pkg/scheduler/predicates.go

```diff
@@ -236,7 +236,7 @@ func NewServiceAffinityPredicate(podLister PodLister, serviceLister ServiceListe
 // - L is listed in the ServiceAffinity object that is passed into the function
 // - the pod does not have any NodeSelector for L
 // - some other pod from the same service is already scheduled onto a minion that has value V for label L
-func (s *ServiceAffinity) CheckServiceAffinity(pod api.Pod, existingPods []api.Pod, node string) (bool, error) {
+func (s *ServiceAffinity) CheckServiceAffinity(pod api.Pod, existingPods []api.Pod, node api.Node) (bool, error) {
```

Comment:
This gets to the discussion I was having before - from this interface level and up, the consumer *does not care* about the UID of the node.  The name of the node is the unique identifier, so there's really no reason to change the assumptions of this interface or the code that depends on it.